client/material/tooltip.tsx

```diff
@@ -0,0 +1,255 @@
+import { shadow2dp } from './shadows'
+import { defaultSpring } from './springs'
+
+export type TooltipPosition = 'left' | 'right' | 'top' | 'bottom'
+
+const NoPointerPortal = styled(Portal)`
+  pointer-events: none;
+`
+
+const TooltipContent = styled.div<{ $position: TooltipPosition }>`
+  ${caption};
+  ${shadow2dp};
+
+  position: relative;
+  min-height: 24px;
+  padding: 4px 8px;
+
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  border: 1px solid rgba(255, 255, 255, 0.36);
+  border-radius: 4px;
+  background-color: ${background900};
+  pointer-events: none;
+
+  &::before {
+    content: '';
+    position: absolute;
+    width: 8px;
+    height: 8px;
+
+    background-color: inherit;
+    border: 1px solid rgba(255, 255, 255, 0.36);
+
+    ${props => {
+      if (props.$position === 'left') {
+        return css`
+          top: 50%;
+          right: 0px;
+
+          transform: translate(50%, -50%) rotate(45deg);
+          border-bottom: none;
+          border-left: none;
+        `
+      } else if (props.$position === 'right') {
+        return css`
+          top: 50%;
+          left: 0px;
+
+          transform: translate(-50%, -50%) rotate(45deg);
+          border-top: none;
+          border-right: none;
+        `
+      } else if (props.$position === 'top') {
+        return css`
+          bottom: 0px;
+          left: 50%;
+
+          transform: translate(-50%, 50%) rotate(45deg);
+          border-top: none;
+          border-left: none;
+        `
+      } else if (props.$position === 'bottom') {
+        return css`
+          top: 0px;
+          left: 50%;
+
+          transform: translate(-50%, -50%) rotate(45deg);
+          border-bottom: none;
+          border-right: none;
+        `
+      } else {
+        return ''
+      }
+    }}
+  }
+`
+
+const NoPointerPopoverContent = styled(PopoverContent)`
+  pointer-events: none;
+`
+
+interface TooltipChildrenWrapperProps {
+  /** The Tooltip text that will be displayed in an `aria-label` attribute. */
+  text: string
+  /** The Tooltip children element(s) that this component will wrap. */
+  children: React.ReactNode
+  /** Class name applied to the children wrapper container element. */
+  className?: string
+  /** Event handler that will be called when a cursor enters the Tooltip target area. */
+  onMouseEnter: (event: React.MouseEvent) => void
+  /** Event handler that will be called when a cursor leaves the Tooltip target area. */
+  onMouseLeave: (event: React.MouseEvent) => void
+}
+
+/**
+ * A helper component that wraps the children that were sent to the Tooltip component, thus ensuring
+ * that the Tooltip component itself always has a single child on which to attach the mouse event
+ * listeners.
+ */
+function TooltipChildrenWrapper({
+  text,
+  children,
+  className,
+  onMouseEnter,
+  onMouseLeave,
+}: TooltipChildrenWrapperProps) {
+  return (
+    <div
+      className={className}
+      aria-label={text}
```

Comment:
Like I mentioned in the preview review, this should probably be customizable, it won't *always* make sense to label the tooltip anchor the same as the tooltip I don't think. You can default it to the tooltip text by doing `ariaLabel = text` in the function params